src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -530,7 +572,9 @@ public bool RemoveTransaction(Keccak? hash)
             bool hasBeenRemoved;
             lock (_locker)
             {
-                hasBeenRemoved = _transactions.TryRemove(hash, out Transaction? transaction);
+                hasBeenRemoved = _transactions.TryRemove(hash, out Transaction? transaction)
+                                 || _blobTransactions.TryRemove(hash, out transaction);
```

Comment:
We have only hash, we don't know if tx SupportsBlobs or not, it is why we are trying to remove it from both collections. Firstly from standard _transactions, as there is more of them, so we will more often succeed with first TryRemove. We can potentially change RemoveTransaction to have tx as an argument instead of hash and then TryRemove only from proper collection